### PR TITLE
Expose iter_size solver option

### LINKED ISSUE #660: Exposing iter_size parameter in DIGITS interface to enable larger training batches
As mentioned in https://github.com/NVIDIA/caffe/issues/16, 0.14 Caffe now has the iter_size option which enables delayed batching and hence larger training batches. It would be great to have this option in DIGITS. Or alternatively, option to manually edit the solver file before training like suggested in https://github.com/NVIDIA/DIGITS/issues/565.

### PR DESCRIPTION
**Pull request by lukeyeager**
*Close #660*
*Replace #675*

Expose `iter_size` solver option (https://github.com/BVLC/caffe/pull/1977) for NVcaffe >= 0.14.

/cc @drozdvadym

### CODE REVIEW

File: digits/templates/models/images/classification/new.html [hunk 1]
```diff
@@ -437,6 +438,17 @@ <h4 style="display:inline-block;">Python Layers</h4>
...
+    $('#iter-size-option').hide();
+    $('#iter_size').prop('disabled', true);
+    {% for fw in frameworks %}
+    {% if fw.can_accumulate_gradients() %}
+    if (fwid == "{{ fw.get_id() }}") {
+        $('#iter_size').prop('disabled', false);
+        $('#iter-size-option').show();
+    }
+    {% endif %}
+    {% endfor %}
```
Comment (gheinrich) [inline digits/templates/models/images/classification/new.html]: or maybe you can extend the structure on lines 414-416 with a `can_accumulate_gradients` field for a more direct access to framework properties?

Comment (lukeyeager) [inline digits/templates/models/images/classification/new.html]: Oh yeah, good call. I also forgot to add this to the `Generic` path ...

File: digits/model/images/classification/views.py [hunk 2]
```diff
@@ -253,23 +253,24 @@ def create():
...
+                        batch_size = form.batch_size.data[0],
+                        batch_accumulation = form.batch_accumulation.data,
+                        val_interval = form.val_interval.data,
+                        pretrained_model = pretrained_model,
+                        crop_size = form.crop_size.data,
+                        use_mean = form.use_mean.data,
+                        network = network,
+                        random_seed = form.random_seed.data,
+                        solver_type = form.solver_type.data,
+                        shuffle = form.shuffle.data,
```
Comment (lukeyeager) [inline digits/model/images/classification/views.py]: Sorry for the whitespace changes. It had to be done at some point.

**Comment by lukeyeager** [comment]
Refactored to use the name `batch_accumulation` instead of `iter_size`.

**Comment by gheinrich** [comment]
That looks good to me - I haven't tested let me know if you want me to.

**Comment by lukeyeager** [comment]
Actually it would be great if you could do a quick smoketest. No need to validate convergence, just check that:

1. The option doesn't appear for Torch
2. It does appear for NVcaffe >= 0.14
3. When you set the value to 2, the generated `solver.prototxt` contains the new value

Assuming all is good, please merge. Thanks!

File: digits/model/tasks/caffe_train.py [hunk 3]
```diff
@@ -426,6 +426,11 @@ def save_files_classification(self):
 
         solver.snapshot_prefix = self.snapshot_prefix
 
+        # Batch accumulation
+        from digits.frameworks import CaffeFramework
+        if CaffeFramework().can_accumulate_gradients():
+            solver.iter_size = self.batch_accumulation
+
```
Comment (gheinrich) [inline digits/model/tasks/caffe_train.py]: I think you need to make this change in `save_files_generic()` too.

**Comment by gheinrich** [comment]
The classification path is working for me - there is a little omission in the generic path.

File: digits/model/tasks/caffe_train.py (hunk 3 above)
Comment (lukeyeager) [inline digits/model/tasks/caffe_train.py]: d'oh

**Pull request merged**